File: python/program/GerbongPenumpang.py
from Gerbong import Gerbong
from Trainmark import Trainmark

class GerbongPenumpang(Gerbong):
    def __init__(self, kode_sarana: str = "", nama_sarana: str = "", berat_sarana: int = 0, tipe_bogie: str = "", jenis_trainmark: Trainmark = None, kapasitas_penumpang: int = 0, kelas_layanan: str = ""):
        # Call parent constructor
        super().__init__(kode_sarana, nama_sarana, berat_sarana, tipe_bogie, jenis_trainmark)
        self._kapasitas_penumpang = kapasitas_penumpang
        self._kelas_layanan = kelas_layanan

    # Method to print GerbongPenumpang information
    def print_info(self):
        print(f"\n=== Info Gerbong Penumpang ({self.get_kode_sarana()}) {self.get_nama_sarana()} ===")
        print(f"Berat Sarana: {self.get_berat_sarana()} kg")
        print(f"Tipe Bogie: {self.get_tipe_bogie()}")
        print(f"Jenis Trainmark: {self.get_jenis_trainmark().getKodeTrainmark()}, Kecepatan Maks: {self.get_jenis_trainmark().getKecepatanMaks()} km/h")
        print(f"Kapasitas Penumpang: {self._kapasitas_penumpang} orang")
        print(f"Kelas Layanan: {self._kelas_layanan}")

    # Getter and Setter methods for kapasitas_penumpang
    def get_kapasitas_penumpang(self) -> int: return self._kapasitas_penumpang
    def set_kapasitas_penumpang(self, kapasitas: int): self._kapasitas_penumpang = kapasitas

    # Getter and Setter methods for kelas_layanan
    def get_kelas_layanan(self) -> str: return self._kelas_layanan
    def set_kelas_layanan(self, kelas: str): self._kelas_layanan = kelas

File: java/program/GerbongPenumpang.java
public class GerbongPenumpang extends Gerbong {
    // Attributes 
    private int kapasitasPenumpang;
    private String kelasLayanan;

    // Default Constructor
    public GerbongPenumpang() {
        super();
        this.kapasitasPenumpang = 0;
        this.kelasLayanan = "";
    }

    // Parameterized Constructor
    public GerbongPenumpang(String kodeSarana, String namaSarana, int beratSarana, String tipeBogie, Trainmark jenisTrainmark, int kapasitasPenumpang, String kelasLayanan) {
        super(kodeSarana, namaSarana, beratSarana, tipeBogie, jenisTrainmark);
        this.kapasitasPenumpang = kapasitasPenumpang;
        this.kelasLayanan = kelasLayanan;
    }

    // Method to print info
    public void printInfo() {
        System.out.println("\n=== Info Gerbong Penumpang (" + this.getKodeSarana() + ") " + this.getNamaSarana() + " ===");
        System.out.println("Berat Sarana: " + this.getBeratSarana() + " kg");
        System.out.println("Tipe Bogie: " + this.getTipeBogie());
        System.out.println("Jenis Trainmark: " + this.getJenisTrainmark().getKodeTrainmark() + ", Kecepatan Maks: " + this.getJenisTrainmark().getKecepatanMaks() + " km/h");
        System.out.println("Kapasitas Penumpang: " + this.kapasitasPenumpang + " orang");
        System.out.println("Kelas Layanan: " + this.kelasLayanan);
    }

    // Getter and Setter for kapasitasPenumpang
    public int getKapasitasPenumpang() {return this.kapasitasPenumpang;}
    public void setKapasitasPenumpang(int kapasitas) {this.kapasitasPenumpang = kapasitas;}

    // Getter and Setter for kelasLayanan
    public String getKelasLayanan() {return this.kelasLayanan;}
    public void setKelasLayanan(String kelas) {this.kelasLayanan = kelas;}
}